### PR TITLE
Feature/guest class setup

### PR DESCRIPTION
**Pull request by saraho1123**
## Is this a fix or a feature?
feature

## What is the change?
Add Guest class with properties and methods.

I also realized as I was working that I could have a parent class for methods that both Guest and Manager use. I added it as Hotel class and it is included in this branch. I made several minor changes to Manager class to reflect adding a parent class.

## What does it fix?
n/a (although there are changes to Manager class)

## Where should the reviewer start?
line 1 of Guest.js
line 1 of Hotel.js
line 1 of Manager.js
line 1 of guest-test.js
line 39 of manager.test.js

index.js has notes for logic moving forward with project
base.scss and index.html have some minor changes and notes for future changes
several changes were made to sample data files in order to make better tests.

## How should this be tested?
run npm test from terminal

## Relevant Tickets:
#1 
#4

### CODE REVIEW

File: src/Guest.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import Hotel from '../src/Hotel';
+
+class Guest extends Hotel {
+  constructor(usersData, roomsData, bookingsData) {
+    super(usersData, roomsData, bookingsData)
+    this.pastBookings = [];
+    this.upcomingBookings = [];
+    this.verySorryMessage = 'There are no available rooms of this time for the date you have picked. We are so very sorry! We love all our guests and really hope to see you very soon! Please click the "Choose New Date" button, or choose a different style of room for this date. '
```
Comment (saraho1123) [inline src/Guest.js:8]: Change time to type!!

Comment (nathanielmillard) [reply src/Guest.js:8]: Nice error/sad path handling!

File: src/Manager.js [hunk 2]
```diff
@@ -1,56 +1,8 @@
+import Hotel from '../src/Hotel';
 
-class Manager {
+class Manager extends Hotel {
   constructor(userData, roomData, bookingsData) {
-    this.users = userData;
```
Comment (nathanielmillard) [inline src/Manager.js:4]: Looks like a pretty big refactor, and glad to see you are working through where you want functional repsonility to lie betwen a parent and child class.

File: src/Guest.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  listVacantRoomsByDate(date) {
+    const bookedRoomNumbers = this.listOccupiedRoomsByDate(date)
+    const availableRoomsOnDate = this.rooms.reduce((available, room) => {
+      !bookedRoomNumbers.includes(room.number) ? available.push(room) : null;
+      return available
+    }, [])
+    return availableRoomsOnDate;
+  }
+
+  filterRoomsByTypeOnDate(type, date) {
```
Comment (atownse) [inline src/Guest.js:54]: this function is a bit lengthy, you could split it up into two

**Review by nathanielmillard — APPROVED** [review]
Over all I have to say things are looking good!
The use of hotel as a parent class that is extended into the two other classes feels like a good way to have shared functionality between Manager and Guest.

File: src/Hotel.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  getSelectedGuestBookings() {
+    this.selectedGuestBookings = this.bookings.filter(booking => {
+      let guestBookings = booking.userID === this.selectedGuest.id;
+      return guestBookings;
+    })
+  }
+
+  bookRoomForGuest(userID, date, roomNumber) { 
+    let bookingData =  { userID: userID, date: date, roomNumber: roomNumber }
+    // call fetchPOST here, I think
```
Comment (atownse) [inline src/Hotel.js:25]: Let me know if this works here, I didn't try this

File: src/Hotel.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    return bookingData;
+  }
+
```
Comment (atownse) [inline src/Hotel.js:28]: Remember to add method that provide revenue for that day

File: src/Guest.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+  listOccupiedRoomsByDate(date) {
+    const bookedRoomsOnDate = this.bookings.filter(booking => {
+      return booking.date === date;
+    })
+    return bookedRoomsOnDate.map(booked => {
+      return booked.roomNumber
+    })
+  }
+
+  listVacantRoomsByDate(date) {
```
Comment (atownse) [inline src/Guest.js:45]: Move this method to parent class

**Review by atownse — APPROVED** [review]

**Pull request merged**